src/transforms/levels.jl

```diff
@@ -29,28 +29,24 @@ Levels(; kwargs...) = throw(ArgumentError("Cannot create a Levels object without
 
 isrevertible(transform::Levels) = true
 
-_categorical(x::AbstractVector, l, o) =
-  categorical(x, levels=l, ordered=o), y -> unwrap.(y)
-
-function _categorical(x::CategoricalArray, l, o)
-  xl, xo = levels(x), isordered(x)
-  revfunc = y -> categorical(y, levels=xl, ordered=xo)
-  categorical(x, levels=l, ordered=o), revfunc
-end
-
 function apply(transform::Levels, table)
   cols = Tables.columns(table)
   names = Tables.columnnames(cols)
   snames = choose(transform.colspec, names)
   ordered = choose(transform.ordered, snames)
-  levels = transform.levels
+  tlevels = transform.levels
   
   results = map(names) do nm
     x = Tables.getcolumn(cols, nm)
     if nm ∈ snames
+      assert_categorical(x)
+      
       o = nm ∈ ordered
-      l = levels[findfirst(==(nm), snames)]
-      return _categorical(x, l, o)
+      l = tlevels[findfirst(==(nm), snames)]
+      y = categorical(x, levels=l, ordered=o)
+      xl, xo = levels(x), isordered(x)
+      revfunc = y -> categorical(y, levels=xl, ordered=xo)
+      return y, revfunc
```

Comment:
```suggestion
      y, revfunc
```

Reply:
This `return` is necessary.

Reply:
Oh so maybe we are missing an else branch in the if statement? I reviewed the code quickly and missed that this was an actual case of early termination.

Reply:
Take a look at the new version of the code, @juliohm.